src/Time/Units.hs

```diff
@@ -85,3 +88,11 @@ convertUnit (Time ratNat) = Time $ ratNat * (ratVal $ divRat (Proxy @unitFrom) (
 -- | Extracts integer part of time unit.
 intPart :: Time unit -> Natural
 intPart (Time n) = numerator n `div` denominator n
+
+-- | Convenient version of 'Control.Concurrent.threadDelay' which takes
+-- any time-unit and operates in any MonadIO.
```

Comment:
`threadDelay` from `Control.Concurrent` takes `Int` while you're converting `Natural` to `Int`. `Natural` is arbitrary precision number. `Int` has upper bound in `9 * 10^18` approximately. Maybe we should add warning in documentation that people should pass bigger time units? :slightly_smiling_face: Well, in reality people won't pass such big numbers. But they accidentally can... And we probably shouldn't fail with exception. When you implement playground, I want you to test behavior with big time units, how big `Natural` converts to `Int` and also (if required) check how `threadDelay` behaves when you pass negative `Int` to it.

Reply:
I figured out that very big `Natural` gives negative number when using convertion to `Int`. But function `threadDelay` from `Concurrent` works without exception on that cases, without any delay. My example was
```
let billionDays = Time 1000000000 :: Day
    let mcs :: Int = fromIntegral $ intPart $ (convertUnit billionDays :: MicroSecond)
    putStrLn $ "Billion days is " ++ show mcs ++ " microseconds"
    Concurrent.threadDelay (-5000000)
    putStrLn "Hmm"
```
and immediate result
```
Billion days is -5833720368547758080 microseconds
Hmm
```

Reply:
@vrom911 Thanks for your investigation! I wasn't aware of such behavior. Well, probably we shouldn't add extra warning to our `threadDelay` function... Users have the same chances to fuck up with `threadDelay` from `Control.Concurrent` module. I guess we didn't introduce any new bugs. Even more, our `threadDelay` is safer. Because unexpected behavior for `threadDelay` can be achieved by passing negative `Int` while with `o-clock` you can't pass negative `Time`. You will get negative time only in case of **really** big time.